Src/Support/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -87,6 +87,28 @@ public static Task<GoogleCredential> GetApplicationDefaultAsync()
         }
 
         /// <summary>
+        /// Returns the Application Default Credentials (ambient credentials) for the application, synchronously.
+        /// </summary>
+        /// <remarks>
+        /// See <see cref="GetApplicationDefaultAsync"/> for more details on the Application Default Credentials. If you
+        /// are writing an application using asynchronous operations, it is encouraged that you use that method. However,
```

Comment:
If we decide to keep this, we need to trim this documentation to highlight what we want developers to know:
- Here's a link to GetApplicationDefaultAsync for what this does.
- You should probably use GetApplicationDefaultAsync.
- This may take a long time.

The next six lines of safety guarantees boil down to "this function will not deadlock", which may go without saying. If we really feel we need to address it, let's just say: "It is okay to call this function from a single-threaded synchronization context, e.g. the Windows Forms UI thread."

The details of *why* it happens to be safe -- including the assumptions we need to ensure are maintained in the future -- should be comments in the implementation.